src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1501,12 +1501,31 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  AppIdURL app_url = policy_manager_->GetNextUpdateUrl(urls);
-  while (!IsUrlAppIdValid(app_url.first, urls)) {
-    app_url = policy_manager_->GetNextUpdateUrl(urls);
+  const uint32_t app_id = GetAppIdForSending();
+  const std::string app_policy_id =
+      application_manager_.application(app_id)->policy_app_id();
+
+  const auto is_appid_or_default = [app_policy_id](const EndpointData& ed) {
+    return helpers::Compare<std::string, helpers::EQ, helpers::ONE>(
+        ed.app_policy_id, policy::kDefaultId, app_policy_id);
+  };
+
+  EndpointUrls::const_iterator app_item =
+      std::find_if(urls.begin(), urls.end(), is_appid_or_default);
+
+  if (app_item == urls.end() || app_item->url.empty()) {
```

Comment:
@ZhdanovP Use the following order in comparisons: `if (urls.end() == app_item)` is equal to `if (rvalue == lvalue)`. 
This way will allow you to avoid hardly recognized errors

Reply:
done